OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUpgradeWebViewFragment.java

```diff
@@ -69,9 +88,38 @@ public void onStop() {
         if (getActivity() != null && authWebView.getWebView() != null) {
             final String url = authWebView.getWebView().getUrl();
             if (url != null && url.contains("checkout/receipt")) {
-                getActivity().finish();
+                onCoursePaymentSuccessful();
             }
+        }
+    }
 
+    public void onCoursePaymentSuccessful() {
+        // Finish activity
+        final Activity activity = getActivity();
+        if( activity != null) {
+            activity.finish();
+        }
+
+        final Bundle arguments = getArguments();
+        if (arguments != null) {
+            final EnrolledCoursesResponse courseData =
```

Comment:
`courseData` can be removed, this is used only to get the `courseId` that we can get through `courseUnit` like the following statement
```
courseUnit.getRoot().getCourseId();
```
**ps:** please remove the passing of `courseData` from the paranet activities/fragment too, if not necessory.

Reply:
We need it when the user will be redirected from the Course Dashboard screen.